///////////////////////////////////////////////////
//         Define Protobuf Syntax Version        //         
///////////////////////////////////////////////////
syntax = "proto3";

///////////////////////////////////////////////////
//              Define Package Name              //         
///////////////////////////////////////////////////
package calculator;

///////////////////////////////////////////////////
//               Define RPC Service              //         
///////////////////////////////////////////////////
service CalculatorService {
    rpc Multiply(MultiplyRequest) returns (MultiplyResponse) {}
    rpc Add(AddRequest) returns (AddResponse) {}
}

///////////////////////////////////////////////////
//              Define Message Type              //         
///////////////////////////////////////////////////
message MultiplyRequest {
    operation_details operation_details = 1;
    int32 first_number = 2;
    int32 second_number = 3;
}

message MultiplyResponse {
    operation_details operation_details = 1;
    int32 result = 2;
}

message AddRequest {
    operation_details operation_details = 1;
    int32 first_number = 2;
    int32 second_number = 3;
}

message AddResponse {
    operation_details operation_details = 1;
    int32 result = 2;
}

message operation_details {
    string operation_name = 1;
    int32 operation_id = 2;
    string operation_start_time = 3;
    string operation_end_time = 4;
}